### ISSUE TITLE
Division by zero error

### ISSUE THREAD

**Issue by constantinpape** [hci-unihd/batchlib#96]
I think the issue is that `n_cells=0` because all were marked as outliers. @imagirom, can we wrap this into the `nan_on_exception` you implemented? 
```
Traceback (most recent call last):
  File "/g/kreshuk/pape/Work/covid/batchlib/batchlib/workflow.py", line 119, in run_workflow
    state = job(folder, **run_kwargs)
  File "/g/kreshuk/pape/Work/covid/batchlib/batchlib/base.py", line 220, in __call__
    self.run(input_files, output_files, **kwargs)
  File "/g/kreshuk/pape/Work/covid/batchlib/batchlib/analysis/cell_level_analysis.py", line 890, in run                                                                                                    
    image_table, image_columns = self.write_image_table(input_files)
  File "/g/kreshuk/pape/Work/covid/batchlib/batchlib/analysis/cell_level_analysis.py", line 727, in write_image_table                                                                                      
    stat_dict = self.get_stat_dict(infected_cell_statistics, control_cell_statistics, bg_keys)
  File "/g/kreshuk/pape/Work/covid/batchlib/batchlib/analysis/cell_level_analysis.py", line 544, in get_stat_dict                                                                                          
    stat_dict['fraction_infected'] = stat_dict['n_infected'] / stat_dict['n_cells']
ZeroDivisionError: division by zero
Traceback (most recent call last):
  File "cell_analysis_workflow.py", line 6, in <module>
    run_cell_analysis(config)
  File "/g/kreshuk/pape/Work/covid/batchlib/batchlib/workflows/cell_analysis.py", line 236, in run_cell_analysis                                                                                           
    ignore_failed_outputs=config.ignore_failed_outputs)
  File "/g/kreshuk/pape/Work/covid/batchlib/batchlib/workflow.py", line 124, in run_workflow
    raise e
  File "/g/kreshuk/pape/Work/covid/batchlib/batchlib/workflow.py", line 119, in run_workflow
    state = job(folder, **run_kwargs)
  File "/g/kreshuk/pape/Work/covid/batchlib/batchlib/base.py", line 220, in __call__
    self.run(input_files, output_files, **kwargs)
  File "/g/kreshuk/pape/Work/covid/batchlib/batchlib/analysis/cell_level_analysis.py", line 890, in run                                                                                                    
    image_table, image_columns = self.write_image_table(input_files)
  File "/g/kreshuk/pape/Work/covid/batchlib/batchlib/analysis/cell_level_analysis.py", line 727, in write_image_table                                                                                      
    stat_dict = self.get_stat_dict(infected_cell_statistics, control_cell_statistics, bg_keys)
  File "/g/kreshuk/pape/Work/covid/batchlib/batchlib/analysis/cell_level_analysis.py", line 544, in get_stat_dict                                                                                          
    stat_dict['fraction_infected'] = stat_dict['n_infected'] / stat_dict['n_cells']
ZeroDivisionError: division by zero
```

**Comment by imagirom** [comment]
Yes, `nan_on_exception(lambda: stat_dict['n_infected'] / stat_dict['n_cells'])()` should do the trick. Not sure if this is the best style..

**Comment by imagirom** [comment]
Should I fix it or would it be easiest for you to just replace the line?

**Comment by constantinpape** [comment]
Yes, I will do it!

**Comment by constantinpape** [comment]
Implemented as part of #95 and it works.

**Comment by constantinpape** [comment]
Thanks @imagirom
